src/main/java/com/redhat/devtools/lsp4ij/settings/ui/JsonTextField.java

```diff
@@ -85,6 +91,16 @@ public void setJsonFilename(@NotNull String jsonFilename) {
         }
     }
 
+    public void associateWithJsonSchema(@NotNull String jsonSchemaContent) {
+        var manager = LSPServerConfigurationJsonSchemaManager.getInstance(project);
+        if (index == null) {
```

Comment:
It doesn't seem like `JsonTextField` should have any concept of how all of this schema stuff works, particularly how the schema manager indexes its schemas. It's just a UI component. It seems like whatever handling is required to derive a schema should happen outside of this component, and the caller should just tell the component the name of a JSON file associated with the correct schema, and then the IDE's JSON schema association features would automatically take care of the rest.

Reply:
Where can I do this code?

Reply:
I add some comments, please tell me if it is better.

Reply:
I still feel like this violates the separation-of-concerns / single responsibility principles. `JsonTextField` should be exactly that, and its only notion of how a JSON schema can be applied is via the filename. The IDE's own JSON schema mappings then take care of actually using the correct JSON schema. How a JSON schema is derived should be external to the component.